### PR TITLE
Deprecate pre 5.6 Worker APIs completely

### PR DESCRIPTION
**Pull request by jjohannes**
- Deprecate old Worker APIs completely and emit deprecation warning
- Update integration tests to use the new API in all places where the old one is not explicitly tested
- Prepare for removal in Gradle 8.0 - follow up: https://github.com/gradle/gradle/issues/15888

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

File: subprojects/workers/src/main/java/org/gradle/workers/internal/IsolationMode.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal;
+
+/**
+ * Isolation mode for workers.
+ */
+public enum IsolationMode {
```
Comment (ghale) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/IsolationMode.java:22]: Having a second (internal) copy of this class could be confusing.  I think we could just get rid of this class altogether now.  It only gets used by `AbstractDaemonCompiler` now to determine what sort of `WorkerFactory` it's been given.  Instead, I wonder if we could introduce a `DaemonCompilerWorkerFactory` (or some such) that wraps the original `WorkerFactory` instance and takes a T spec as its argument.  Then, `AbstractDaemonCompiler` just gets one of these and calls `getWorker(spec)` and it takes care of creating a `WorkerRequirement` and calling the underlying `WorkerFactory`.  We could inject it from the compiler factories.  Does that make sense?  Happy to chat about it if not.

**Review by ghale — COMMENTED** [review]
Good stuff @jjohannes!  I left a comment about one possible improvement we could make here to finish cleaning this up.

**Pull request merged**